core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -605,7 +608,19 @@ public List<TableColumnsMetadata> listTableColumns(Session session, QualifiedTab
                             }
                         }
                         catch (RuntimeException e) {
-                            if (!(e instanceof TrinoException trinoException) || !trinoException.getErrorCode().equals(UNSUPPORTED_TABLE_TYPE.toErrorCode())) {
+                            boolean silent = false;
+                            if (e instanceof TrinoException trinoException) {
+                                ErrorCode errorCode = trinoException.getErrorCode();
+                                silent = errorCode.equals(UNSUPPORTED_TABLE_TYPE.toErrorCode()) ||
+                                        // e.g. table deleted concurrently
+                                        errorCode.equals(NOT_FOUND.toErrorCode()) ||
+                                        // e.g. Iceberg/Delta table being deleted concurrently resulting in failure to load metadata from filesystem
+                                        errorCode.getType() == EXTERNAL;
+                            }
+                            if (silent) {
+                                log.debug(e, "Failed to get metadata for table: %s", objectName);
+                            }
+                            else {
                                 log.warn(e, "Failed to get metadata for table: %s", objectName);
```

Comment:
why not use `debug` by default?

Reply:
debug = ignored (and not logged by default)

i don't want to leave no trace in logs of some unforeseen problems like NPE